frontend/leaderboard/src/Challenges.re

```diff
@@ -63,6 +65,79 @@ let getSnarkWorkCreatedByUser = blocks => {
   blocks |> calculateProperty(calculateSnarkWorkCount);
 };
 
+module SnarkFeesCollectedMap = Map.Make(String);
+
+let snarkFeesCollected = blocks => {
+  Array.fold_left(
+    (map, block: Types.NewBlock.t) => {
+      Array.fold_left(
+        (map, snarkJob: Types.NewBlock.snarkJobs) => {
+          SnarkFeesCollectedMap.update(
+            snarkJob.prover,
+            feeCount =>
+              switch (feeCount) {
+              | Some(feeCount) => Some(Int64.add(feeCount, snarkJob.fee))
+              | None => Some(snarkJob.fee)
+              },
+            map,
+          )
+        },
+        map,
+        block.data.newBlock.snarkJobs,
+      )
+    },
+    SnarkFeesCollectedMap.empty,
+    blocks,
+  );
+};
+
+let max = (a, b) => {
+  a > b ? a : b;
+};
+
+let highestSnarkFeeCollected = blocks => {
+  Array.fold_left(
+    (map, block: Types.NewBlock.t) => {
+      Array.fold_left(
+        (map, snarkJob: Types.NewBlock.snarkJobs) => {
```

Comment:
Is this metric supposed to be snark work completed or snark work included in the block? Looking at the docs on Block.snarkJobs, it seems to represent "Completed snark works".  Seems to come from `Staged_ledger_diff.completed_works` in the code. Any idea if this is the list of included work or just completed? I guess the fact that it's attached to a block might indicate the former (in which case I should maybe improve the docs a little)
cc @bkase

Reply:
We should definitely improve the docs -- if the snark work is included in a block and has an author associated with it it is by definition completed (incomplete work won't have an associated fee or creator attached to it)

Reply:
Since there's a `.prover` on it -- this seems right